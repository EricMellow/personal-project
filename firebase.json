{
  "hosting": {
    "public": "/",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**",
      "keys.js"
    ]
  }
}
