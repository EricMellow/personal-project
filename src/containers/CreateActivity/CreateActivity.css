.create {
  align-items: center;
  background-repeat: no-repeat;
  background-size: cover;
  background: url(../../assets/bricks1.jpg); 
  display: flex;
  flex-direction: column;
  height: 85vh;
  justify-content: center;
}

.create-form{
  align-items: center;
  background-color: rgba(28, 141, 196, .9);
  display: flex;
  flex-direction: column;
  height: 70%;
  justify-content: center;
  width: 50%;
}

.input {
  background: none;
  border-radius: 5px;
  border: 3px solid white;
  color: white;
  font-size: 2rem;
  height: 3rem;
  margin: 20px;
  padding-left: 10px;
  width: 75%;
}

.add-btn {
  background: none;
  border-radius: 5px;
  border: 2px solid white;
  color: white;
  font-size: 2rem;
  height: 3rem;
  margin: 20px;
  width: 75%;
}

.add-btn:disabled {
  background: rgba(255, 255, 255, 0.15);
  border: 2px solid rgb(67, 119, 144);
  color: rgb(67, 119, 144);
}

*, *:before, *:after {
  box-sizing: inherit;
}

.message {
  font-size: 2rem
}

form p {
  color: white;
  font-size: 1.2rem;
  margin: 0;
}