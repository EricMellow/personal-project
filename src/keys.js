export const distanceMatrixKey = '';

export const firebaseKey = '';