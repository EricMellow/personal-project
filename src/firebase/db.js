import { db } from './firebase';

// User API

export const doCreateUser = (id, username, email, zipcode) =>
  db.ref(`users/${id}`).set({
    username,
    email,
    zipcode
  });

export const doCreateActivity = (address, duration, lat, lng, type) => 
  db.ref('actions/').push({
    address,
    type,
    duration,
    lat,
    lng
  })

export const doGetAllActivities = () => 
  db.ref('actions/').on('value', snapshot => snapshot.val())

// export const onceGetUsers = () =>
//   db.ref('users').once('value');