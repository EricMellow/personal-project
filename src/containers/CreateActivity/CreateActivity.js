import React, { Component } from "react";
import './CreateActivity.css';
import { firebaseKey } from "../../keys";

export class CreateActivity extends Component {
  constructor(props) {
    super(props);
    this.state = {
      address: '',
      type: '',
      duration: ''
    };
  }

  handleInput = (event) => {
    event.preventDefault();
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  storeActivity = async (event) => {
    event.preventDefault();
    const location = await this.getLocation();
  }
  
  getLocation = async () => {
    const response = await fetch(`https://maps.googleapis.com/maps/api/geocode/json?address=${this.state.address}&key=${firebaseKey}`)
    const location = await response.json();
    return location.results[0].geometry.location;
  }

  render() {
    const {
      type,
      address,
      duration
    } = this.state;

    const formsFilled =
      address === '' ||
      type === '' ||
      duration === '';

    return (
      <form className="create" onSubmit={event => this.storeActivity(event)}>
        <input
          placeholder="Event Type"
          name="type"
          className="input type-input"
          value={type}
          onChange={event => this.handleInput(event)}
          type="text"
        />
        <input
          placeholder="Address"
          name="address"
          className="input address-input"
          value={address}
          onChange={event => this.handleInput(event)}
          type="text"
        />
        <input
          placeholder="Duration"
          name="duration"
          className="input duration-input"
          value={duration}
          onChange={event => this.handleInput(event)}
          type="text"
        />
        <button
          type="submit"
          className="add-btn"
          disabled={formsFilled}
        >
          Add Activity
        </button>

        {/* {error && <p>{error}</p>} */}
      </form>
    );
  }
}

export default CreateActivity;