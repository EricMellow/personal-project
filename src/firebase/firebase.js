import { firebaseKey } from '../keys';
import * as firebase from 'firebase';
const key = firebaseKey;
// Initialize Firebase
var config = {
  apiKey: key,
  authDomain: "personal-project-1527122816740.firebaseapp.com",
  databaseURL: "https://personal-project-1527122816740.firebaseio.com",
  projectId: "personal-project-1527122816740",
  storageBucket: "",
  messagingSenderId: "182263229649"
};

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

const db = firebase.database();
const auth = firebase.auth();

export { db, auth };