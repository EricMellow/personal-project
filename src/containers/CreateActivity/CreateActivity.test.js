import React from "react";
import { CreateActivity } from "./CreateActivity";
import { shallow } from "enzyme";
import { db } from '../../firebase';
import { firebaseKey } from "../../keys";


describe('CreateActivity', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<CreateActivity />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should start with a default state', () => {
    const wrapper = shallow(<CreateActivity />);
    const expected = {
      address: '',
      type: '',
      duration: '',
      message: false
    };

    expect(wrapper.state()).toEqual(expected);
  });

  describe('handleInput', () => {

    it('should set the appropriate key in state with the event value', () => {
      const wrapper = shallow(<CreateActivity />);
      const mockEvent = {
        preventDefault: jest.fn(),
        target: {
          name: 'type',
          value: 'frolf'
        }
      };
      const initialState = {
        address: '',
        type: '',
        duration: '',
        message: false
      };
      const expected = {
        address: '',
        type: 'frolf',
        duration: '',
        message: false
      };

      expect(wrapper.state()).toEqual(initialState);
      wrapper.instance().handleInput(mockEvent);
      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe('storeActivity', () => {
    let wrapper;
    let mockEvent;

    beforeEach(() => {
      wrapper = shallow(<CreateActivity />);
      mockEvent = {
        preventDefault: jest.fn()
      };
      wrapper.instance().getLocation = jest.fn().mockImplementation(() => {
        return {
          lat: 12345,
          lng: 54321
        };
      });
      wrapper.setState({
        address: '221 B Baker St.',
        type: 'Investigation',
        duration: '16 hours'
      });
      db.doCreateActivity = jest.fn();
    });

    it('should call db.doCreateActivity with the correct arguments', async () => {
      const expectedAddress = '221 B Baker St.';
      const expectedDuration = '16 hours';
      const expectedLat = 38.8372452;
      const expectedLng = -97.617204;
      const expectedType = 'INVESTIGATION';
      const expectedTime = 23;
      Date.now = jest.fn().mockImplementation(() => 23);

      await wrapper.instance().storeActivity(mockEvent);
      expect(db.doCreateActivity).toHaveBeenCalledWith(
        expectedAddress, 
        expectedDuration, 
        expectedLat, 
        expectedLng, 
        expectedType, 
        expectedTime
      );
    });

    it('should reset state to the default settings', async () => {
      const initialState = {
        address: '221 B Baker St.',
        type: 'Investigation',
        duration: '16 hours',
        message: false
      };
      const expected = {
        address: '',
        type: '',
        duration: '',
        message: true
      };

      expect(wrapper.state()).toEqual(initialState);
      await wrapper.instance().storeActivity(mockEvent);
      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe('resetInputFields', () => {
    it('should reset state to its default state', () => {
      const wrapper = shallow(<CreateActivity />);
      const initialState = {
        address: '221 B Baker St.',
        type: 'Investigation',
        duration: '16 hours',
        message: false
      };
      const expected = {
        address: '',
        type: '',
        duration: '',
        message: false
      };
      wrapper.setState(initialState);

      expect(wrapper.state()).toEqual(initialState);
      wrapper.instance().resetInputFields();
      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<CreateActivity />);
    });
    
    it('should call storeActivity onSubmit of the form', () => {
      wrapper.instance().storeActivity = jest.fn();

      wrapper.find('.create-form').simulate('submit');
      expect(wrapper.instance().storeActivity).toHaveBeenCalled();
    });

    it('should call handleInput onChange of the type input', () => {
      wrapper.instance().handleInput = jest.fn();

      wrapper.find('.type-input').simulate('change');
      expect(wrapper.instance().handleInput).toHaveBeenCalled();
    });

    it('should call handleInput onChange of the address input', () => {
      wrapper.instance().handleInput = jest.fn();

      wrapper.find('.address-input').simulate('change');
      expect(wrapper.instance().handleInput).toHaveBeenCalled();
    });

    it('should call handleInput onChange of the duration input', () => {
      wrapper.instance().handleInput = jest.fn();

      wrapper.find('.duration-input').simulate('change');
      expect(wrapper.instance().handleInput).toHaveBeenCalled();
    });
  });

});