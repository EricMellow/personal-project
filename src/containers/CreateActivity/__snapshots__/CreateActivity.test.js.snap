// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateActivity should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateActivity />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="create-form"
        onSubmit={[Function]}
>
        <input
                className="input type-input"
                name="type"
                onChange={[Function]}
                placeholder="Event Type"
                type="text"
                value=""
        />
        <input
                className="input address-input"
                name="address"
                onChange={[Function]}
                placeholder="Address"
                type="text"
                value=""
        />
        <p>
                *Enter as address, city, state
        </p>
        <input
                className="input duration-input"
                name="duration"
                onChange={[Function]}
                placeholder="Duration"
                type="number"
                value=""
        />
        <p>
                *Rounded to the nearest hour
        </p>
        <button
                className="add-btn"
                disabled={true}
                type="submit"
        >
                Add Activity
        </button>
</form>,
      "className": "create",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="input type-input"
            name="type"
            onChange={[Function]}
            placeholder="Event Type"
            type="text"
            value=""
/>,
          <input
            className="input address-input"
            name="address"
            onChange={[Function]}
            placeholder="Address"
            type="text"
            value=""
/>,
          <p>
            *Enter as address, city, state
</p>,
          <input
            className="input duration-input"
            name="duration"
            onChange={[Function]}
            placeholder="Duration"
            type="number"
            value=""
/>,
          <p>
            *Rounded to the nearest hour
</p>,
          <button
            className="add-btn"
            disabled={true}
            type="submit"
>
            Add Activity
</button>,
          false,
        ],
        "className": "create-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input type-input",
            "name": "type",
            "onChange": [Function],
            "placeholder": "Event Type",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input address-input",
            "name": "address",
            "onChange": [Function],
            "placeholder": "Address",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "*Enter as address, city, state",
          },
          "ref": null,
          "rendered": "*Enter as address, city, state",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input duration-input",
            "name": "duration",
            "onChange": [Function],
            "placeholder": "Duration",
            "type": "number",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "*Rounded to the nearest hour",
          },
          "ref": null,
          "rendered": "*Rounded to the nearest hour",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Activity",
            "className": "add-btn",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Add Activity",
          "type": "button",
        },
        false,
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="create-form"
          onSubmit={[Function]}
>
          <input
                    className="input type-input"
                    name="type"
                    onChange={[Function]}
                    placeholder="Event Type"
                    type="text"
                    value=""
          />
          <input
                    className="input address-input"
                    name="address"
                    onChange={[Function]}
                    placeholder="Address"
                    type="text"
                    value=""
          />
          <p>
                    *Enter as address, city, state
          </p>
          <input
                    className="input duration-input"
                    name="duration"
                    onChange={[Function]}
                    placeholder="Duration"
                    type="number"
                    value=""
          />
          <p>
                    *Rounded to the nearest hour
          </p>
          <button
                    className="add-btn"
                    disabled={true}
                    type="submit"
          >
                    Add Activity
          </button>
</form>,
        "className": "create",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="input type-input"
              name="type"
              onChange={[Function]}
              placeholder="Event Type"
              type="text"
              value=""
/>,
            <input
              className="input address-input"
              name="address"
              onChange={[Function]}
              placeholder="Address"
              type="text"
              value=""
/>,
            <p>
              *Enter as address, city, state
</p>,
            <input
              className="input duration-input"
              name="duration"
              onChange={[Function]}
              placeholder="Duration"
              type="number"
              value=""
/>,
            <p>
              *Rounded to the nearest hour
</p>,
            <button
              className="add-btn"
              disabled={true}
              type="submit"
>
              Add Activity
</button>,
            false,
          ],
          "className": "create-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input type-input",
              "name": "type",
              "onChange": [Function],
              "placeholder": "Event Type",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input address-input",
              "name": "address",
              "onChange": [Function],
              "placeholder": "Address",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "*Enter as address, city, state",
            },
            "ref": null,
            "rendered": "*Enter as address, city, state",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input duration-input",
              "name": "duration",
              "onChange": [Function],
              "placeholder": "Duration",
              "type": "number",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "*Rounded to the nearest hour",
            },
            "ref": null,
            "rendered": "*Rounded to the nearest hour",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Activity",
              "className": "add-btn",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Add Activity",
            "type": "button",
          },
          false,
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
